text/0000-infered-types.md

```diff
@@ -0,0 +1,174 @@
+- Feature Name: Inferred Types
+- Start Date: 2023-06-06
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+
+# Summary
+[summary]: #summary
+
+This RFC introduces a feature allowing the base type of enums and structs to be inferred in certain contexts including but not limited to function calls. The syntax is as follows `_::Variant`.
+
+
+# Motivation
+[motivation]: #motivation
+
+Using large libraries usually requires developers to import large amounts of traits, structures, and enums. To just call a function on an implementation can take upwards of 3 imports. One way developers have solved this is by importing everything from specific modules. Importing everything has its own problems like trying to guess where imports are actually from. Developers need a low compromise solution to solve this problem and that comes in the form of inferred types.
+
+Swift has had a system to infer types since around 2014. Its system has been praised by many developers around the world. It’s system is as follows:
+```swift
+enum EnumExample {
+   case variant
+   case variant2
+}
+
+
+example(data: .variant);
+```
+
+This RFC’s intent was to create something similar to the system already tried and tested in swift. Additionally, the underscore is already used to imply type’s lifetimes so it runs consistent with the rust theme.
+
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+
+Implied types basically replace the path to a type with `_`. They can be used in places where strong typing exists. This includes function calls and function returns. You can think about the `_` as expanding into the type’s name.
```

Comment:
> They can be used in places where strong typing exists.

Rust is strongly typed. I think this should be more precise, like saying "where type inference has enough information to deduce the base type" or something.

Alternatively, it may be better to exhaustively specify positions (like we do with coercion sites: https://doc.rust-lang.org/reference/type-coercions.html) where it's possible to use this syntax.